InvasivesBC/ViewControllers/BaseViewController.swift

```diff
@@ -0,0 +1,237 @@
+//
+//  BaseViewController.swift
+//  ipad
+//
+//  Created by Amir Shayegh on 2019-10-29.
+//  Copyright © 2019 Amir Shayegh. All rights reserved.
+//
+
+import UIKit
+import DatePicker
+
+protocol InputDelegate: class {
+    func showDropdownDelegate(items: [DropdownModel], on view: UIView, callback: @escaping (_ selection: DropdownModel?) -> Void)
+    func showDatepickerDelegate(on view: UIView, initialDate: Date?, minDate: Date?, maxDate: Date?, callback: @escaping (_ date: Date?) -> Void)
+    func showTimePickerDelegate(on view: UIView, initialTime: Time?, completion: @escaping (_ time: Time?) -> Void)
+    func showOptionsDelegate(options: [OptionType], on button: UIButton, completion: @escaping (_ option: OptionType) -> Void)
+}
+
+class BaseViewController: UIViewController, Theme ,InputDelegate {
```

Comment:
Implement delegate in an extension and in a different file. For private methods, you can marked them as internal. This not part of API code, so strict access level control is not required.